### PR TITLE
Add dataloader pre-trained embeddings support to Merlin Models

### LINKED ISSUE #1068: Create a PretrainedEmbeddings block and integrate it with InputBlockV2.
Prototype API

```python
input_block = InputBlockV2(schema, 
                   categorical: Union[Tag,Block] = Tag.CATEGORICAL, 
                   continuous: Union[Tag,Block] = Tag.CONTINUOUS,  
                    pretrained_embeddings: Union[Tag,Block] = Tag.EMBEDDINGS, ...)

def PretrainedEmbeddings(schema, 
                                      projection: Optional[Union[int,dict,Block,Layer,Dict[str, Layer]]] = None, 
                                      normalization: Union[str, Layer] = None,
                                      sequence_combiner = "mean") -> ParallelBlock:
```

- Include some common normalization options: e.g. L2 normalization, standardization?

If `PretrainedEmbeddings.projection` is:
- int or dict[int], we create MLPBlock for each branch
- Dict[str, Layer] or ParallelBlock, the keys should match the feature names and we just connect the pre-trained embeddings with the projection layers
- Layer (not ParallelBlock) -  we need to use an updated version of the MapValues layer, that take a `shared` arg and in the build() we clone it for each branch

## Starting point
- The `Embeddings` function in MM TF API
- The PyT, where `Embeddings` is a class extending `ParallelBlock` 

## Testing
- Create tests on how to combine/aggregate pre-trained embeddings with other features and also how to aggregate sequential features:
- Agregating 3D a sequence feature into 2D: `SequenceAggregator`: "max", "min", "sum", "mean"
- Aggregating multiple features: `ConcatFeatures`, `ElementwiseSum`, `ElementWiseMultiply`

### LINKED ISSUE #1070: Adapt SequenceMasking to support pre-trained embeddings

### LINKED ISSUE #1071: Convert sequential pre-trained embeddings into 3d ragged tensor representation and ensure SequenceTransform accepts 3D ragged tensor

### LINKED ISSUE #1072: Update models ranking models (DLRM, DCN) to support PretrainedEmbeddings with fixed dim projection.

### LINKED ISSUE #1073: Ensure that `BroadcastFeatures` support expanding 2D pre-trained embeddings (contextual features) to concat with 3D sequence features

### PR DESCRIPTION
**Pull request by gabrielspmoreira**
Fixes #1070, Fixes #1071, Fixes #1068, Fixes #1072, Fixes #1073

### Goals :soccer:
- Add support to pre-trained embeddings provided by the `EmbeddingOperator` transform from Merlin dataloader:  https://github.com/NVIDIA-Merlin/Merlin/issues/211

### Tasks
- [x] #1071
- [x] #1068
- [x] #1073
- [x] #1070
- [x] #1072

### Implementation Details :construction:
- Creates the `PretrainedEmbeddings` block that takes pre-trained embedding features and optionally projects them to a dim with a linear layer, applies a sequence aggregator and normalizes (e.g. with l2-norm). All of these options are configurable
- The `InputBlockV2` was changed to accept an optional `pretrained_embeddings` argument, which by default selects features tagged with the EMBEDDING tag,
- `PrepareListFeatures` was changed to define the shape of the last dim the pre-trained embeddings provided by the Loader, as that dim is None in graph mode.

### Testing Details :mag:
- Created many tests demonstrating how pre-trained embeddings can be used with models like DLRM, DCN, sequential Transformer models with BroadcastFeatures and with causal and masked language modeling SequenceMasking classes
- I took this opportunity also to try and **speed up many tests** by reducing drastically the cardinality of categorical features in some dataset schemas used for synthetic data generation. Many tests had to be updated to match the new cardinalities.

### CODE REVIEW

File: merlin/models/tf/transforms/regularization.py [hunk 1]
```diff
@@ -32,11 +32,39 @@ def __init__(self, **kwargs):
 
     def call(self, inputs: Union[tf.Tensor, TabularData], axis: int = -1, **kwargs):
         if isinstance(inputs, dict):
-            inputs = {key: tf.linalg.l2_normalize(inp, axis=axis) for key, inp in inputs.items()}
+            inputs = {key: self._l2_norm(inp, axis=axis) for key, inp in inputs.items()}
         else:
-            inputs = tf.linalg.l2_normalize(inputs, axis=axis)
+            inputs = self._l2_norm(inputs)
```
Comment (sararb) [inline merlin/models/tf/transforms/regularization.py:37]: we might need to specify the axis parameter in this line

File: merlin/models/tf/transforms/regularization.py [hunk 2]
```diff
@@ -32,11 +32,39 @@ def __init__(self, **kwargs):
...
 
+    def _l2_norm(
+        self,
+        inputs: Union[tf.Tensor, tf.SparseTensor, tf.RaggedTensor],
+        epsilon: float = 1e-12,
+        axis: int = -1,
+    ) -> Union[tf.Tensor, tf.SparseTensor, tf.RaggedTensor]:
+        """Computes L2-norm for a given axis, typically axis = -1.
+        Equivalent to tf.linalg.l2_normalize(), but that function
+        does not support tf.RaggedTensor
```
Comment (sararb) [inline merlin/models/tf/transforms/regularization.py:49]: Good to know, thanks for the fix!

**Review by sararb — APPROVED** [review]
Thank you for the PR @gabrielspmoreira! This looks good to me, I've just left some minor comments.

File: merlin/models/tf/inputs/embedding.py [hunk 3]
```diff
@@ -625,6 +622,95 @@ def _get_dim(col, embedding_dims, infer_dim_fn):
...
+    ParallelBlock
+        Returns a parallel block with a branch for each pre-trained embedding
+    """
+
+    tables = {}
+
+    for col in schema:
+        table_name = col.name
+
+        tables[table_name] = NoOp()
```
Comment (marcromeyn) [inline merlin/models/tf/inputs/embedding.py:679]: Should this be configurable to allow users to do things like normalization?

**Review by marcromeyn — APPROVED** [review]
